python/pyspark/sql/pandas/types.py

```diff
@@ -190,10 +191,19 @@ def _check_series_localize_timestamps(s, timezone):
     """
     from pyspark.sql.pandas.utils import require_minimum_pandas_version
     require_minimum_pandas_version()
-
+    from pandas import Series, DatetimeTZDtype
+    from pandas.api.types import is_datetime64_dtype
     from pandas.api.types import is_datetime64tz_dtype
     tz = timezone or _get_local_timezone()
     # TODO: handle nested timestamps, such as ArrayType(TimestampType())?
+    # Check if its Array of TimeStamp
+    if datatype == ArrayType.__name__:
```

Comment:
does it work for nested + nested array of timestamps?

Reply:
Currently only one level of nesting is handled . So only ArrayType<TimeStamp> will work . Also this is what being suggested in Jira (as the immediate need)